internal/controller/pkg/revision/establisher.go

```diff
@@ -94,7 +101,7 @@ type currentDesired struct {
 // Establish checks that control or ownership of resources can be established by
 // parent, then establishes it.
 func (e *APIEstablisher) Establish(ctx context.Context, objs []runtime.Object, parent v1.PackageRevision, control bool) ([]xpv1.TypedReference, error) { // nolint:gocyclo
-	allObjs := []currentDesired{}
+	allObjs := make(chan currentDesired, len(objs))
```

Comment:
```suggestion
	allObjs := make(chan currentDesired, maxConcurrentReconciles)
```

Would this make sense, or is there an advantage to letting the entire set of objects into the buffer?

Reply:
since we're validating all resources before syncing them, we have to buffer the results otherwise the `g.Wait()` below will end up blocking forever.